src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -0,0 +1,214 @@
+                            Assert.Equal(1, activity.Baggage.Count());
+
+                            using (Activity activity1 = aSource.StartActivity("AllDataRequestedActivity1"))
+                            {
+                                Assert.NotNull(activity1);
+                                Assert.True(activity1.IsAllDataRequested);
+                                Assert.Equal(2, counter);
+
+                                Assert.Equal(0, activity1.Links.Count());
+                                Assert.Equal(0, activity1.Events.Count());
+                                Assert.True(object.ReferenceEquals(activity1, activity1.AddLink(new ActivityLink(new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.None, null)))));
+                                Assert.True(object.ReferenceEquals(activity1, activity1.AddEvent(new ActivityEvent("e1"))));
+                                Assert.Equal(1, activity1.Links.Count());
+                                Assert.Equal(1, activity1.Events.Count());
+                            }
+                            Assert.Equal(1, counter);
+                        }
+
+                        Assert.Equal(0, counter);
+                    }
+                }
+            }).Dispose();
+        }
+
+        [Fact]
+        public void TestActivitySourceAttachedObject()
+        {
+            RemoteExecutor.Invoke(() => {
+                // All Activities created through the constructor should have same source.
+                Assert.True(object.ReferenceEquals(new Activity("a1").Source, new Activity("a2").Source));
+                Assert.True(string.IsNullOrEmpty(new Activity("a3").Source.Name));
+                Assert.Equal(new Version(), new Activity("a4").Source.Version);
+
+                using (ActivitySource aSource = new ActivitySource("SourceToTest", new Version(1, 2, 3, 4)))
+                {
+                    //Ensure at least we have a listener to allow Activity creation
+                    using (IDisposable listener1 = ActivitySource.AddActivityListener(
+                                    (activitySource) => object.ReferenceEquals(aSource, activitySource),
+                                    (source, name, kind, context, tags, links) => ActivityDataRequest.AllData,
+                                    (source, name, kind, parentId, tags, links) => ActivityDataRequest.AllData,
+                                    activity => Assert.NotNull(activity), activity => Assert.NotNull(activity)))
+                    {
+                        using (Activity activity = aSource.StartActivity("ActivityToTest"))
+                        {
+                            Assert.True(object.ReferenceEquals(aSource, activity.Source));
+                        }
+                    }
+                }
+            }).Dispose();
+        }
+
+        public void Dispose() => Activity.Current = null;
+    }
+}
```

Comment:
A few other tests that might be useful:
1. Confirm listeners receive start/stop events for Activities created via public constructor
2. Confirm multiple listeners with different requests generate highest request
3. Confirm that created activity started via the many arg forms of StartActivity has correct name, parentId, links, kind, tags and id format. It should be possible to do both W3C and HierarchialId formats using overload where parentId is a string.

Reply:
I added all listed tests